src/habitat.cr

```diff
@@ -31,10 +31,17 @@ class Habitat
     # Habitat.raise_if_missing_settings!
     # ```
     def self.raise_if_missing_settings!
+      {% # https://github.com/crystal-lang/crystal/pull/14490
+
+ if compare_versions(Crystal::VERSION, "1.13.0-dev") >= 0
+   nil_type_id = "::Nil".id
```

Comment:
I'm wary of this change because https://github.com/luckyframework/habitat/issues/88 seems like it's saying this might not work or always work.

I think I found a way to update this to work for current and future versions. Like he said with path having the leading "::", if you call [`Path#names`](https://crystal-lang.org/api/1.12.2/Crystal/Macros/Path.html#names%3AArrayLiteral%28MacroId%29-instance-method) it should separate based on those and the leading "::" is dropped. `Namespaced::Class` becomes `[Namespaced, Class]` and `::Nil` is just `[Nil]`. To use that in this scenario, the condition becomes

```crystal
!setting[:decl].type.types.any?(&.names.includes?(Nil.id)))
```

Never used `names` before, but it seems like it works well and we don't need any crystal version conditionals.

Reply:
Oh, nice find! I don't think I've seen that either, but definitely worth a shot since this felt pretty hacky as is.